File: files/2026-03-22/CSEC 22032026.R
library(tidyverse)
library(readxl)

path <- "2026-03-22/Challenge 108.xlsx"
input1 <- read_excel(path, range = "B2:B4")
input2 <- read_excel(path, range = "B6:B9")
test <- read_excel(path, range = "D2:D14")

result = expand.grid(input1$Mentors, input2$Staff) %>%
  as.matrix() %>%
  t() %>%
  c()

all.equal(result, test$Session)

File: files/2026-03-22/CSEC 22032026.py
import pandas as pd
from itertools import chain

path = "2026-03-22/Challenge 108.xlsx"
input1 = pd.read_excel(path, usecols="B", nrows = 2, skiprows = 1).iloc[:, 0].tolist()
input2 = pd.read_excel(path, usecols="B", nrows = 3, skiprows = 5).iloc[:, 0].tolist()
test = pd.read_excel(path, usecols="D", nrows = 13, skiprows = 1)

result = list(chain.from_iterable((a, b) for b in input2 for a in input1))

print(result == test.values.flatten().tolist())
## [1] TRUE